<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Anthony Devaux, Boris Hejblum" />

<meta name="date" content="2018-07-26" />

<title>Userguide to the tcgsaseq R package</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Userguide to the <code>tcgsaseq</code> R package</h1>
<h4 class="author"><em>Anthony Devaux, Boris Hejblum</em></h4>
<h4 class="date"><em>2018-07-26</em></h4>


<div id="TOC">
<ul>
<li><a href="#overview-of-tcgsaseq"><span class="toc-section-number">1</span> Overview of tcgsaseq</a></li>
<li><a href="#getting-started-using-tcgsaseq"><span class="toc-section-number">2</span> Getting started using tcgsaseq</a><ul>
<li><a href="#rna-seq-expression-data"><span class="toc-section-number">2.1</span> RNA-seq expression data</a></li>
<li><a href="#model-covariates-matrix"><span class="toc-section-number">2.2</span> Model covariates matrix</a></li>
<li><a href="#design-matrix"><span class="toc-section-number">2.3</span> Design matrix</a></li>
<li><a href="#gene-set-object"><span class="toc-section-number">2.4</span> Gene set object</a></li>
</ul></li>
<li><a href="#gene-set-analysis-of-rna-seq-data-tcgsa_seq-function-from-baduel-et-al.-2016"><span class="toc-section-number">3</span> Gene set analysis of RNA-seq data : tcgsa_seq function (from <em>Baduel et al., 2016</em>)</a><ul>
<li><a href="#data-preparation"><span class="toc-section-number">3.1</span> Data preparation</a><ul>
<li><a href="#rna-seq-expression-data-1"><span class="toc-section-number">3.1.1</span> RNA-seq expression data</a></li>
<li><a href="#covariates-matrix"><span class="toc-section-number">3.1.2</span> Covariates matrix</a></li>
<li><a href="#design-matrix-1"><span class="toc-section-number">3.1.3</span> Design matrix</a></li>
<li><a href="#gene-set-object-1"><span class="toc-section-number">3.1.4</span> Gene set object</a></li>
</ul></li>
<li><a href="#tcgsa_seq-function"><span class="toc-section-number">3.2</span> tcgsa_seq function</a></li>
</ul></li>
<li><a href="#gene-wise-differential-analysis-of-rna-seq-data-varseq-function-from-dorr-et-al.-2015"><span class="toc-section-number">4</span> Gene-wise differential analysis of RNA-seq data : varseq function (from <em>Dorr et al., 2015</em>)</a><ul>
<li><a href="#data-preparation-1"><span class="toc-section-number">4.1</span> Data preparation</a><ul>
<li><a href="#rna-seq-expression-data-2"><span class="toc-section-number">4.1.1</span> RNA-seq expression data</a></li>
<li><a href="#covariates-matrix-1"><span class="toc-section-number">4.1.2</span> Covariates matrix</a></li>
<li><a href="#design-matrix-2"><span class="toc-section-number">4.1.3</span> Design matrix</a></li>
</ul></li>
<li><a href="#varseq-function"><span class="toc-section-number">4.2</span> varseq function</a></li>
</ul></li>
<li><a href="#references"><span class="toc-section-number">5</span> References</a></li>
</ul>
</div>

<div id="overview-of-tcgsaseq" class="section level1">
<h1><span class="header-section-number">1</span> Overview of tcgsaseq</h1>
<p>The <code>tcgsaseq</code> <code>R</code> package is a principled, model-free, and efficient method for detecting longitudinal changes in RNA-seq gene sets defined <em>a priori</em>. The method identifies those gene sets whose expression varies over time. <code>tcgsaseq</code> is based on an original variance component score test accounting for both covariates and heteroscedasticity without assuming any specific parametric distribution for the (transformed) counts. For more details, check the <a href="https://academic.oup.com/biostatistics/article/18/4/589/3065599">publish article in Biostatistics</a>.</p>
<p>To illustrate the methods, we use the data arising from <em>Baduel et al., 2016</em> and <em>Dorr et al., 2015</em> to measure time-course gene set expression of the plant <em>Arabidopsis arenosa</em> and kidney transplant patients respectively.</p>
</div>
<div id="getting-started-using-tcgsaseq" class="section level1">
<h1><span class="header-section-number">2</span> Getting started using tcgsaseq</h1>
<p>Required inputs :</p>
<ul>
<li>The RNA-seq expression data matrix</li>
<li>The covariate matrix containing the model covariates</li>
<li>The design matrix containing the variables to be tested</li>
<li><strong>In case of gene set analysis (tcgsa_seq function)</strong>, the gene set object</li>
</ul>
<div id="rna-seq-expression-data" class="section level2">
<h2><span class="header-section-number">2.1</span> RNA-seq expression data</h2>
<p>This matrix contains the RNA-seq expression data (in cells) for each genes (in rows) of each samples (in columns). The data can be raw RNA-seq counts or preprocessed expressions. In case of raw RNA-seq counts, the data is automatically transform to <strong>normalize data on reads per million (RPM)</strong> in <code>tcgsa_seq</code> or <code>varseq</code> function. For the preprocessed expressions, <code>preprocessed</code> argument should be <code>TRUE</code> in the <code>tcgsa_seq</code> or <code>varseq</code> function’s option. The dimensions of this matrix should be n x G with n number of samples and G number of genes.</p>
</div>
<div id="model-covariates-matrix" class="section level2">
<h2><span class="header-section-number">2.2</span> Model covariates matrix</h2>
<p>This matrix contains the variables which compose the nonparametric regression. The first column should be an intercept (n-length vector of 1) and can be follow-up by unique variables and interaction variables (combinations of several variables). All variables should be type as numeric in this matrix. The dimensions of this matrix should be n x p for n samples and p covariates variables.</p>
</div>
<div id="design-matrix" class="section level2">
<h2><span class="header-section-number">2.3</span> Design matrix</h2>
<p>This matrix contained the variables to be tested. The dimensions should be n x K for the n samples and the K variables to be tested.</p>
</div>
<div id="gene-set-object" class="section level2">
<h2><span class="header-section-number">2.4</span> Gene set object</h2>
<p>A gene set is a group of genes either sharing the same biological function. It enables to detect different gene expression and seems to be more powerful than a gene-by-gene analysis. Several definitions of groups have been made, in particular here we will focus on the following:</p>
<ul>
<li>Chaussabel’s modules (<em>Chaussabel et al., 2008</em>)</li>
<li>Gene Ontology database (<em>Ashburner et al., 2000</em>)</li>
<li>Kyoto Encyclopedia of Genes and Genomes database (<em>Kanehisa et Goto, 2000</em>)</li>
</ul>
<p>The gene set object is a <code>gmt</code> format containing:</p>
<ul>
<li>the name of the gene set</li>
<li>the description of the gene set, such as biological function</li>
<li>the list of genes inside each gene set</li>
</ul>
<p>One can either use already existing <code>gmt</code> objects, or build their own. To self-build your <code>gmt</code> object, you have to prepare a <code>.gmt</code> file. This file format is the tab delimited file which can be created with <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29">this helpful website</a> from the Broad Institute. In this file, one row represents one gene set with:</p>
<ul>
<li>Column 1: name of the gene set</li>
<li>Column 2: description of the gene set</li>
<li>Remaining columns: list of genes included in the gene set</li>
</ul>
<p>Next, to import the <code>.gmt</code> file into R, you need to run the <code>GSA.read.gmt()</code> function from the <code>GSA</code> package. More details on the <code>GSA</code> help package.</p>
</div>
</div>
<div id="gene-set-analysis-of-rna-seq-data-tcgsa_seq-function-from-baduel-et-al.-2016" class="section level1">
<h1><span class="header-section-number">3</span> Gene set analysis of RNA-seq data : tcgsa_seq function (from <em>Baduel et al., 2016</em>)</h1>
<p><code>tcgsa_seq</code> is the main function of <code>tcgsaseq</code> package to compute gene set analysis. To give an example, we try to figure out the different flowering strategies between the two populations of <em>Arabidopsis arenosa</em> (denoted KA and TBG) and specially which gene sets have a different activation between those populations, adjusted for the plant age and the cold exposure (also called “Vernalization”).</p>
<div id="data-preparation" class="section level2">
<h2><span class="header-section-number">3.1</span> Data preparation</h2>
<div id="rna-seq-expression-data-1" class="section level3">
<h3><span class="header-section-number">3.1.1</span> RNA-seq expression data</h3>
<p>First of all, to compute the <code>tcgsa_seq</code> function, our data should be either raw RNA-seq counts or preprocessed expressions. For our RNA-seq expressions, we have to apply only the log2 transformation to have preprocessed expressions :</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">expr_log2 &lt;-<span class="st"> </span><span class="kw">log2</span>(expr_norm_corr<span class="op">+</span><span class="dv">1</span>)</code></pre></div>
</div>
<div id="covariates-matrix" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Covariates matrix</h3>
<p>To build the covariates matrix from the regression model, several variables should be created :</p>
<ul>
<li>an intercept : <code>Intercept</code> (n-length vector of 1)</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design<span class="op">$</span>Intercept &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>,<span class="kw">nrow</span>(design))</code></pre></div>
<ul>
<li>interaction(s) : <code>Vernalized_Population</code> and <code>AgeWeeks_Population</code> from <code>Vernalized</code>, <code>Population</code> and <code>AgeWeeks</code> variables.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design<span class="op">$</span>Vernalized_Population &lt;-<span class="st"> </span><span class="kw">ifelse</span>(design<span class="op">$</span>Population<span class="op">==</span><span class="st">&quot;KA&quot;</span><span class="op">&amp;</span>design<span class="op">$</span>Vernalized,<span class="dv">1</span>,<span class="dv">0</span>)
design<span class="op">$</span>AgeWeeks_Population &lt;-<span class="st"> </span><span class="kw">ifelse</span>(design<span class="op">$</span>Population<span class="op">==</span><span class="st">&quot;KA&quot;</span>,design<span class="op">$</span>AgeWeeks,<span class="dv">0</span>)</code></pre></div>
<p>The covariates matrix (named <code>model_covar</code>) contains the different variables which form the regression model according to the question. It must contained an intercept (n-length vector of 1) to work.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model_covar &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">as.matrix</span>(design[, <span class="kw">c</span>(<span class="st">&quot;Intercept&quot;</span>,
                    <span class="st">&quot;Vernalized&quot;</span>, <span class="st">&quot;AgeWeeks&quot;</span>, <span class="st">&quot;Vernalized_Population&quot;</span>, <span class="st">&quot;AgeWeeks_Population&quot;</span>),
                    <span class="dt">drop =</span> <span class="ot">FALSE</span>]), <span class="dv">2</span>, as.numeric)
<span class="kw">head</span>(model_covar)</code></pre></div>
<pre><code>##      Intercept Vernalized AgeWeeks Vernalized_Population
## [1,]         1          0        3                     0
## [2,]         1          0        3                     0
## [3,]         1          0        3                     0
## [4,]         1          0        3                     0
## [5,]         1          0        3                     0
## [6,]         1          0        3                     0
##      AgeWeeks_Population
## [1,]                   3
## [2,]                   3
## [3,]                   3
## [4,]                   0
## [5,]                   0
## [6,]                   0</code></pre>
</div>
<div id="design-matrix-1" class="section level3">
<h3><span class="header-section-number">3.1.3</span> Design matrix</h3>
<p>Design matrix is named for the matrix containing the different variables to be tested. To test the difference between the two populations of the plant, we keep the variable representing the population, but we should transform it as numeric.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design<span class="op">$</span>PopulationKA &lt;-<span class="st"> </span><span class="kw">ifelse</span>(design<span class="op">$</span>Population<span class="op">==</span><span class="st">&quot;KA&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>)
vartest &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(design[, <span class="kw">c</span>(<span class="st">&quot;PopulationKA&quot;</span>), <span class="dt">drop =</span> <span class="ot">FALSE</span>])
<span class="kw">head</span>(vartest)</code></pre></div>
<pre><code>##      PopulationKA
## [1,]            1
## [2,]            1
## [3,]            1
## [4,]            0
## [5,]            0
## [6,]            0</code></pre>
</div>
<div id="gene-set-object-1" class="section level3">
<h3><span class="header-section-number">3.1.4</span> Gene set object</h3>
<p>The gene set object include 2 gene sets from Gene Ontology database (<em>Ashburner et al., 2000</em>) plus 3 self-built gene sets by the authors.</p>
<p>In the case of our study, we used two gene sets from our <code>.gmt</code> object : * <code>VR_TBG</code> for the vernalized TBG * <code>DE_KAvsTBG</code> for the top 1% differentially expressed genes between KA and TBG</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">geneset &lt;-<span class="st"> </span>baduel_gmt<span class="op">$</span>genesets[<span class="kw">c</span>(<span class="st">&quot;VR_TBG&quot;</span>,<span class="st">&quot;DE_KAvsTBG&quot;</span>)]</code></pre></div>
</div>
</div>
<div id="tcgsa_seq-function" class="section level2">
<h2><span class="header-section-number">3.2</span> tcgsa_seq function</h2>
<p>The <code>tcgsa_seq</code> function provides the result of variance component score test for longitudinal gene set analysis. To run this function, it requires :</p>
<ul>
<li><code>y</code> : the RNA-seq expression matrix</li>
<li><code>x</code> : the model covariates matrix</li>
<li><code>phi</code> : the design matrix containing the variables to be tested</li>
<li><code>genesets</code> : the list of genes (named by index or rownames of <code>y</code>) of each gene sets. The value of <code>genesets</code> argument could be <code>NULL</code> in case of no gene sets have been defined (i.e gene-wise analysis).</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">KAvsTBG &lt;-<span class="st"> </span><span class="kw">tcgsa_seq</span>(<span class="dt">y =</span> expr_log2, <span class="dt">x =</span> model_covar, <span class="dt">phi =</span> vartest,
                    <span class="dt">genesets =</span> geneset, <span class="dt">doPlot =</span> <span class="ot">FALSE</span>,
                    <span class="dt">preprocessed =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Computing the weights... 
## Bandwith computed.
## Done!</code></pre>
<p><code>pvals</code> value provides the raw and ajusted p-values for each gene sets selected in <code>genesets</code> argument. The adjusted p-values are computed according to <code>padjust_methods</code> argument (defaut option <code>BH</code> for Benjamini-Hochberg procedure for controlling the FDR).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">KAvsTBG<span class="op">$</span>pvals</code></pre></div>
<pre><code>##            rawPval adjPval
## VR_TBG       0.928   0.928
## DE_KAvsTBG   0.011   0.022</code></pre>
<p>The <code>DE_KAvsTBG</code> gene set is significant rather than <code>VR_TBG</code> gene set (at a 5% threshold). In other words, we found that the gene set <code>DE_KAvsTBG</code> have a significant different activation between the two populations KA and TBG, adjusted for plant age and the cold exposure.</p>
</div>
</div>
<div id="gene-wise-differential-analysis-of-rna-seq-data-varseq-function-from-dorr-et-al.-2015" class="section level1">
<h1><span class="header-section-number">4</span> Gene-wise differential analysis of RNA-seq data : varseq function (from <em>Dorr et al., 2015</em>)</h1>
<p><code>varseq</code> is the main function of <code>tcgsaseq</code> package to compute genewise analysis. To compute this function, the required inputs are the same as <code>tcgsa_seq</code> function except for <code>genesets</code> object.</p>
<p>To illustrate this function, we use the dataset from <em>Dorr et al., 2015</em>. Gene expression was measured in the peripheral blood mononuclear cells of 32 kidney transplant patients at 4 time points: before transplantation, 1 week after transplantation, 3 months after transplantation and 6 months after transplantation. We try to figure out linear change in gene expression over time while adjusting for patient’s age and gender on 32 kidney transplant patients.</p>
<div id="data-preparation-1" class="section level2">
<h2><span class="header-section-number">4.1</span> Data preparation</h2>
<p>We import the RNA-seq expression data and variables matrix from <a href="https://conservancy.umn.edu/">the University of Minnesota Digital Conservancy (UDC) website</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">FPKMs &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">read.delim</span>(<span class="st">&quot;http://conservancy.umn.edu/bitstream/handle/11299/177631.1/FPKMs.txt&quot;</span>, <span class="dt">row.names=</span><span class="dv">1</span>))
clinical &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;http://conservancy.umn.edu/bitstream/handle/11299/177631.1/clinical.txt&quot;</span>)
<span class="kw">head</span>(clinical)</code></pre></div>
<pre><code>##        ID month SID      age gender
## 1 Sample1  0.00   2 60.92813      1
## 2 Sample2  0.00  22 38.04244      0
## 3 Sample3  0.25  22 38.04244      0
## 4 Sample4  3.00  22 38.04244      0
## 5 Sample5  0.00  23 33.16359      1
## 6 Sample6  0.25  23 33.16359      1</code></pre>
<div id="rna-seq-expression-data-2" class="section level3">
<h3><span class="header-section-number">4.1.1</span> RNA-seq expression data</h3>
<p><code>FPKMs</code> contains the raw RNA-seq expression data and we transformed them into reads per million.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cs_fpkm &lt;-<span class="st"> </span><span class="kw">colSums</span>(FPKMs)
TPMs &lt;-<span class="st"> </span>(FPKMs<span class="op">/</span>cs_fpkm)<span class="op">*</span><span class="dv">10</span><span class="op">^</span><span class="dv">6</span>
log2TPMs &lt;-<span class="st"> </span><span class="kw">log2</span>(TPMs <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)</code></pre></div>
</div>
<div id="covariates-matrix-1" class="section level3">
<h3><span class="header-section-number">4.1.2</span> Covariates matrix</h3>
</div>
<div id="design-matrix-2" class="section level3">
<h3><span class="header-section-number">4.1.3</span> Design matrix</h3>
<p>We have to test changes in gene expression over time, so we built the design matrix from the time variable <code>month</code> in the dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phi &lt;-<span class="st"> </span><span class="kw">cbind</span>(clinical[, <span class="st">&quot;month&quot;</span>])
<span class="kw">head</span>(phi)</code></pre></div>
<pre><code>##      [,1]
## [1,] 0.00
## [2,] 0.00
## [3,] 0.25
## [4,] 3.00
## [5,] 0.00
## [6,] 0.25</code></pre>
</div>
</div>
<div id="varseq-function" class="section level2">
<h2><span class="header-section-number">4.2</span> varseq function</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">indiv &lt;-<span class="st"> </span><span class="kw">cbind</span>(clinical[, <span class="st">&quot;SID&quot;</span>])</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_varseq &lt;-<span class="st"> </span><span class="kw">varseq</span>(<span class="dt">exprmat =</span> y, <span class="dt">covariates =</span> x, <span class="dt">variables2test =</span> phi,
                     <span class="dt">preprocessed =</span> <span class="ot">TRUE</span>,
                     <span class="dt">sample_group =</span> indiv)</code></pre></div>
</div>
</div>
<div id="references" class="section level1">
<h1><span class="header-section-number">5</span> References</h1>
<p>Agniel D, Hejblum BP, (2017) Variance component score test for time-course gene set analysis of longitudinal RNA-seq data. <em>Biostatistics</em> 18(4):589-604.</p>
<p>Ashburner M, Ball CA, Blake JA, Botstein D, Butler H, Cherry JM, et al., (2000) Gene Ontology: tool for the unification of biology. <em>Nat Genet</em> 25(1):25-9.</p>
<p>Baduel P, Arnold B, Weisman CM, Hunter B et Bomblies K, (2016) Habitat-associated life history and stress-tolerance variation in <em>Arabidopsis arenosa</em>. <em>Plant Physiology</em> <strong>171</strong>, 437-451.</p>
<p>Chaussabel D, Quinn C, Shen J, Patel P, Glaser C, Baldwin N, et al., (2008) A Modular Analysis Framework for Blood Genomics Studies: Application to Systemic Lupus Erythematosus. <em>Immunity</em> 29(1):150-64.</p>
<p>Dorr C, Wu B, Guan W, Muthusamy A, Sanghavi K, Schladt DP et al., (2015). Differentially expressed gene transcripts using RNA sequencing from the blood of immunosuppressed kidney allograft recipients. <em>PLoS ONE</em> <strong>10</strong>, 1-14.</p>
<p>Kanehisa M, Goto S, (2000) KEGG: Kyoto Encyclopedia of Genes and Genomes. <em>Nucleic Acids Res</em> 28(1):27-30.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
